modules/core/src/smithy4s/schema/Schema.scala

```diff
@@ -79,7 +79,7 @@ sealed trait Schema[A]{
   final def refined[B]: PartiallyAppliedRefinement[A, B] = new PartiallyAppliedRefinement[A, B](this)
 
   final def getDefault: Option[Document] =
-      this.hints.get(smithy.api.Default).map(_.value)
+      DefaultValueSchemaVisitor(this).getDefault
```

Comment:
Should the `DefaultValueSchemaVisitor` be held in a lazy val or something like that?

Reply:
I think you have the wrong way to go at it. When you talked about using a SchemaVisitor, I was thinking of producing the `A`, not the Document. The visitor should be used in `getDefaultValue`, not `getDefault` (unless there's something I've not thought of).

Reply:
https://github.com/disneystreaming/smithy4s/pull/792